systemdspawner/systemdspawner.py

```diff
@@ -157,16 +158,14 @@ def __init__(self, *args, **kwargs):
 
         systemd_version = systemd.get_systemd_version()
         if systemd_version is None:
-            self.log.warning(
-                "Failed to parse systemd version from 'systemctl --version'"
-            )
+            warnings.warn("Failed to parse systemd version from 'systemctl --version'")
         elif systemd_version < SYSTEMD_REQUIRED_VERSION:
             self.log.critical(
                 f"systemd version {SYSTEMD_REQUIRED_VERSION} or higher is required, version {systemd_version} is used"
             )
             sys.exit(1)
         elif systemd_version < SYSTEMD_LOWEST_RECOMMENDED_VERSION:
-            self.log.warning(
+            warnings.warn(
```

Comment:
warnings _almost_ always want a `stacklevel` (usually but not always 2), but I think this is actually a case where it wouldn't be helpful, so 👍 here.